### PR TITLE
Check hippo health

### PR DESCRIPTION
**Pull request by FrankYang0529**
resolves https://github.com/fermyon/spin/issues/562

```
> ~/Project/spin/target/debug/spin deploy -f spin.toml --deploy-existing-bindle
Error: Hippo server http://localhost:5309/ is unhealthy
# follow nomad tutorial to start nomad
> ~/Project/spin/target/debug/spin deploy -f spin.toml --deploy-existing-bindle
Deployed spin-hello-world version 1.0.0+qb632d35
Available Routes:
  hello: http://spin-deploy.spin-hello-world.hippofactory.io/hello
    A simple component that returns hello.
```

### CODE REVIEW

File: src/commands/deploy.rs [hunk 1]
```diff
@@ -295,6 +297,19 @@ impl DeployCommand {
...
     }
+
+    async fn check_hippo_healthz(&self) -> Result<()> {
+        let hippo_base_url = url::Url::parse(&self.hippo_server_url)?;
+        let hippo_healthz_url = hippo_base_url.join("/healthz")?;
+        let result = reqwest::get(hippo_healthz_url.to_string())
+            .await?
+            .text()
+            .await?;
+        if result == "Unhealthy" {
```
[the remark this answers is not in the document]
Comment (dicej) [reply src/commands/deploy.rs:308]: I agree with @lann and would also recommend using [error_for_status](https://docs.rs/reqwest/latest/reqwest/struct.Response.html#method.error_for_status), e.g.:

```rust
        let result = reqwest::get(hippo_healthz_url.to_string())
            .await?
            .error_for_status()?
            .text()
            .await?;
```

[the remark this answers is not in the document]
Comment (FrankYang0529) [reply src/commands/deploy.rs]: Hi @dicej and @lann, thanks for the suggestion. If we use `error_for_status`, it looks like we don't need to check `result` text.

```
> curl -k http://localhost:5309/healthz
Unhealthy%

> ~/Project/spin/target/debug/spin deploy -f spin.toml --deploy-existing-bindle
Error: HTTP status server error (503 Service Unavailable) for url (http://localhost:5309/healthz)

# start nomad

> ~/Project/spin/target/debug/spin deploy -f spin.toml --deploy-existing-bindle
Deployed spin-hello-world version 1.0.0+qb632d35
Available Routes:
  hello: http://spin-deploy.spin-hello-world.hippofactory.io/hello
    A simple component that returns hello.
```

[the remark this answers is not in the document]
Comment (lann) [reply src/commands/deploy.rs]: I believe Hippo still returns 200 with an `Unhealthy` status: https://github.com/deislabs/hippo/issues/889

[the remark this answers is not in the document]
Comment (FrankYang0529) [reply src/commands/deploy.rs]: Thanks for pointing it out. I add the check for `result != "Healthy"` back.

**Review by dicej — APPROVED** [review]

**Review by lann — APPROVED** [review]
Looks good to me. I'd like @michelleN to sanity-check the behavior.

**Comment by bacongobbler** [comment]
FYI in the latest version of hippo, /healthz will now return a 503 if something is unhealthy.

**Review by michelleN — APPROVED** [review]

**Comment by michelleN** [comment]
Going to merge and then we can update.

**Pull request merged**